### PR TITLE
Avoid extra latency potentially happening between `Poll` calls

### PR DESCRIPTION
**Pull request by guizmaii**
Related to work done here: 
- https://github.com/zio/zio-kafka/pull/661

I didn't manage to find a way to test that this "extra latency" is not stacked anymore 🤔

@svroonland I wonder if there's not a risk of generating/stacking a lot of `Poll` in the Stream 🤔
I wonder if using a `Queue.dropping[Command.Poll](1)` wouldn't be a good idea.
At any time, I think that we only need to have 1 or 0 Poll command: if we have one, at the next iteration of the loop, it'll consume it from the queue and will execute the corresponding action. If 0, it'll do nothing.
If the queue was containing 1 Poll command and we're emitting a new one, we're replacing the old one with the new one, so we don't waste memory and we have no risks of stacking too many of them (which could potentially lead to an OOM)

WDYT? 🤔

### CODE REVIEW

**Comment by svroonland** [comment]
> I wonder if there's not a risk of generating/stacking a lot of Poll in the Stream..
Yes, the same thought had crossed my mind this weekend.

File: zio-kafka/src/main/scala/zio/kafka/consumer/diagnostics/DiagnosticEvent.scala [hunk 1]
```diff
@@ -25,4 +26,6 @@ object DiagnosticEvent {
     final case class Assigned(partitions: Set[TopicPartition]) extends Rebalance
     final case class Lost(partitions: Set[TopicPartition])     extends Rebalance
   }
+
+  final case class RunloopEvent(command: Command) extends DiagnosticEvent
```
Comment (svroonland) [inline zio-kafka/src/main/scala/zio/kafka/consumer/diagnostics/DiagnosticEvent.scala:30]: Not a huge fan of this, it means exposing one of the Runloop's internals. Shall we at least document that it's interface is unstable, so we can change the Commands a bit more freely if we need?

Comment (guizmaii) [reply zio-kafka/src/main/scala/zio/kafka/consumer/diagnostics/DiagnosticEvent.scala:30]: Agreed. TBH, I'm not clear what this `Diagnostics` thing is for

**Review by svroonland — DISMISSED** [review]

**Review by svroonland — APPROVED** [review]

**Comment by erikvanoosten** [comment]
> > I wonder if there's not a risk of generating/stacking a lot of Poll in the Stream..
> 
> Yes, the same thought had crossed my mind this weekend.

I assumed it already worked this way, but now I see it doesn't :)
Has this already been addressed? I can't find this change (if it was done already).

**Comment by svroonland** [comment]
Not yet, I'm experimenting a bit with some changed polling behavior though.

**Pull request merged**